### ISSUE TITLE
.NET throw System.ExecutionEngineException on Step Into 2D Array with JMC disabled.

### ISSUE THREAD

**Issue by chuckries** [dotnet/runtime#79415]
### Description

In Visual Studio Debugger, with JMC disabled, if an attempt is made to step into a 2D array accessor, the runtime throws System.ExecutionEngineException. The step in should become a step over. It may seem strange to step into the array accessor, but imagine the following code:

```cs
int[,] grid = new grid[5, 5];
DoSomething(grid[0, 0]);
```
If you attempt to step into the method `DoSomething`, the debugger will attempt to step into the array accessor and the exception will be thrown.

### Reproduction Steps

1. In Visual Studio, create a new .NET Console Application
2. Pate the following code:
```cs
    class Program
    {
        static void Main(string[] args)
        {
            int[,] grid = new int[5, 5];
            int i = grid[0, 0]; // BP here, JMC off, step in
        }
    }
```
3. Set a breakpoint on the commented line
4. Start debugging with F5
5. Once the breakpoint is hit, Step In (F11)

### Expected behavior

The Step In should simply become a Step Over.

### Actual behavior

The runtime throws System.ExecutionEngineException.

### Regression?

This does not repro on .NET Framework, but it reproed against .NET 6.0.11 and .NET Core 3.1.31

### Known Workarounds

N/A

### Configuration

- Windows x64
- Repros in both VS 2022 and 2019
- Rerpos against .NET 6.0 and .NET Core 3.1

### Other information

Debugging through VS, it looks like we create an non-JMC ICorDebugStepper and issue the Step In with our computed range. We immediately get ICorDebugManagedCallback2::Exception(...) without ever receiving StepComplete

**Comment by AaronRobinsonMSFT** [comment]
I believe this was fixed in https://github.com/dotnet/runtime/pull/65947. @chuckries can you try .NET 7 and see if it reproduces?

**Comment by tommcdon** [comment]
Nice catch @AaronRobinsonMSFT!  I verified that the issue is indeed fixed in 7.0.  Since the issue is fixed in 7.0, I'll go ahead and close this issue.  Please let us know if this is blocking 6.0 development then we can take this in for consideration for 6.0 servicing.
